### ISSUE TITLE
Ray Tracing: Ambiguity in the description about "geometryIndex"

### ISSUE THREAD

**Issue by alek314** [KhronosGroup/Vulkan-Docs#1068]
https://github.com/KhronosGroup/Vulkan-Docs/blob/194a7f4d0d552e78d6f1a2fa2ae0ddbe0215090a/chapters/VK_NV_ray_tracing/raytracing.txt#L217

section 33.3.2 reads:
"Geometries refer to a triangle or axis-aligned bounding box."
Does the above sentence imply that geometryIndex is the same as gl_PrimitiveID?

An alternative definition could be an index into "pGeometries" entry of VkAccelerationStructureInfoNV.

Am I missing something? Thank you.

**Comment by ewerness-nv** [comment]
The value returned by gl_PrimitiveID is within a geometry, not an entire instance. I'll be sure we clarify that.

**Comment by dgkoch** [comment]
Proposed spec change (in the definition of PrimitiveId): 
```
In an intersection, any-hit, or closest hit shader, it will contain the
-index of the triangle or bounding box being processed.
+index within the geometry of the triangle or bounding box being processed.
```

**Comment by alek314** [comment]
For me, the root source of confusion is the following code, which comes from [https://developer.nvidia.com/rtx/raytracing/vkray](url):
```
  struct GeometryInstance
  {
    VkBuffer     vertexBuffer;
    uint32_t     vertexCount;
    VkDeviceSize vertexOffset;
    VkBuffer     indexBuffer;
    uint32_t     indexCount;
    VkDeviceSize indexOffset;
    glm::mat4x4  transform;
  };
```

This struct looks very like VkGeometryInstanceNV in the spec:

```
struct VkGeometryInstanceNV {
    float          transform[12];
    uint32_t       instanceCustomIndex : 24;
    uint32_t       mask : 8;
    uint32_t       instanceOffset : 24;
    uint32_t       flags : 8;
    uint64_t       accelerationStructureHandle;
};
```
But the code from the sample mistakenly mix things from TLAS and BLAS together, which is misleading. When I try to add more than one instance to the TLAS based on the sample code, things get very confusing.

I think the official sample is very important for people to understand to spec.But there seems to be no way to leave a comment there. Maybe someone can fix that as well ?😄 

Thank you.

**Comment by dgkoch** [comment]
I'm not familiar with that sample, but the VkGeometryInstanceNV structure is a description of how an instance in the TLAS references a BLAS and how that information is laid out in a VkBuffer that is read by the GPU. 
It doesn't look like the GeometryInstance structure you quoted is directly related. I'll pass the confusion along to the blog authors though.

**Comment by alek314** [comment]
thank you very much

**Comment by oddhack** [comment]
@dgkoch's fix has been included in the 1.1.127 spec update.
